src/main/java/com/packagename/myapp/views/customComponents/UserForm.java

```diff
@@ -0,0 +1,157 @@
+import com.packagename.myapp.services.LoginService;
+import com.vaadin.flow.component.Key;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.datepicker.DatePicker;
+import com.vaadin.flow.component.dialog.Dialog;
+import com.vaadin.flow.component.html.H2;
+import com.vaadin.flow.component.listbox.ListBox;
+import com.vaadin.flow.component.notification.Notification;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.radiobutton.RadioButtonGroup;
+import com.vaadin.flow.component.textfield.EmailField;
+import com.vaadin.flow.component.textfield.PasswordField;
+import com.vaadin.flow.component.textfield.TextField;
+
+public class UserForm extends Dialog {
+
+    private final LoginService loginService;
+    private final UserRepository userRepository;
+    private User user;
+    private Runnable onclose;
+
+    public UserForm(LoginService loginService, UserRepository userRepository, User user, Runnable onclose) {
+        this.loginService = loginService;
+        this.userRepository = userRepository;
+        this.user = user;
+        this.onclose = onclose;
+        init();
+    }
+
+    void init() {
+        H2 title = new H2("Add a new user");
+        TextField newUsernameField = new TextField("Username *", "username");
+        EmailField newEmailField = new EmailField("Email", "email");
+        TextField newNameField = new TextField("Name *", "name");
+        TextField newSurnameField = new TextField("Surname *", "surname");
+        TextField newPhoneNumber = new TextField("Phone Number", "phone number");
+        PasswordField newPasswordField = new PasswordField("Password *", "password");
+        PasswordField newConfirmField = new PasswordField("Confirm your password *", "confirm your password");
+        DatePicker newDateOfBirthField = new DatePicker("Date of birth");
+        ListBox<UserRole> newUserRoleListBox = new ListBox<>();
+        newUserRoleListBox.setItems(UserRole.TEACHER, UserRole.STUDENT);
+        newUserRoleListBox.setValue(UserRole.STUDENT);
+        RadioButtonGroup<String> newUserIsAdmin = new RadioButtonGroup<>();
+        newUserIsAdmin.setLabel("Give this user Admin privileges?");
+        newUserIsAdmin.setItems("Yes", "No");
+        newUserIsAdmin.setValue("No");
+
+
+        Button addUserButton = new Button("Add user", event -> {
+
+            user.setUsername(newUsernameField.getValue());
+            user.setEmail(newEmailField.getValue());
+            user.setName(newNameField.getValue());
+            user.setSurname(newSurnameField.getValue());
+            user.setPhoneNumber(newPhoneNumber.getValue());
+            user.setPassword(HashingService.hashThis(newPasswordField.getValue()));
+            user.setBirthDay(newDateOfBirthField.getValue().toString());
+            if (newUserIsAdmin.getValue().equals("Yes")) {
```

Comment:
If statement can be replaced with a boolean variable:
user.setAdmin(newUserIsAdmin.getValue().equals("Yes"));